sktime/contrib/critical_difference.py

```diff
@@ -0,0 +1,600 @@
+# -*- coding: utf-8 -*-
+"""Standalone function to plot critical difference diagram."""
+
+__author__ = ["Svea Meyer"]
+
+import numpy as np
+from scipy.stats import rankdata, find_repeats, distributions
+import math
+from sktime.utils.validation._dependencies import _check_soft_dependencies
+
+_check_soft_dependencies("matplotlib")
+import matplotlib.pyplot as plt
+
+
+def _check_friedman(n_strategies, n_datasets, ranked_data, alpha):
+    """
+    Check whether Friedman test is significant.
+
+    Larger parts of code copied from scipy.
+
+    Arguments
+    ---------
+    n_strategies : int
+      number of strategies to evaluate
+    n_datasets : int
+      number of datasets classified per strategy
+    ranked_data : np.array (shape: n_strategies * n_datasets)
+      rank of strategy on dataset
+
+    Returns
+    -------
+    is_significant : bool
+      Indicates whether strategies differ significantly in terms of performance
+      (according to Friedman test).
+    """
+    if n_strategies < 3:
+        raise ValueError(
+            "At least 3 sets of measurements must be given for Friedmann test, got{}.".format(
+                n_strategies
+            )
+        )
+
+    # calculate c to correct chisq for ties:
+    ties = 0
+    for i in range(n_datasets):
+        replist, repnum = find_repeats(ranked_data[i])
+        for t in repnum:
+            ties += t * (t * t - 1)
+    c = 1 - ties / (n_strategies * (n_strategies * n_strategies - 1) * n_datasets)
+
+    ssbn = np.sum(ranked_data.sum(axis=0) ** 2)
+    chisq = (
+        12.0 / (n_strategies * n_datasets * (n_strategies + 1)) * ssbn
+        - 3 * n_datasets * (n_strategies + 1)
+    ) / c
+    p = distributions.chi2.sf(chisq, n_strategies - 1)
+    if p < alpha:
+        is_significant = True
+    else:
+        is_significant = False
+    return is_significant
+
+
+def plot_critical_difference(
```

Comment:
this is ok-ish, though I think what the function body should be doing is only plotting.
There is some testing logic in there and clique computation, which I'd suggest to move out to separate functions.
Similarly, one could think about factoring out the raw plotting (only positions, cliques etc passed), and have a main function which does everything together - that would be cleaner and follow the "separation of concerns" pattern better.